sfxclient/sfxclient.go

```diff
@@ -290,9 +291,19 @@ func (s *Scheduler) ReportOnce(ctx context.Context) error {
 		s.previousDatapoints = datapoints
 		return datapoints
 	}()
+	s.prependPrefix(datapoints)
 	return s.Sink.AddDatapoints(ctx, datapoints)
 }
 
+// Add prefix to metrics if specified in scheduler
+func (s *Scheduler) prependPrefix(datapoints []*datapoint.Datapoint) {
+	if s.Prefix != "" {
+		for _, datapoint := range datapoints {
+			datapoint.Metric = fmt.Sprintf("%s%s", s.Prefix, datapoint.Metric)
```

Comment:
everything looks good now. Just something to consider (only if you want to),

i did benchmark comparing fmt.Sprintf and strings.Builder below are the result:

```
BenchmarkSprintf-12         	    3000	  16667999 ns/op	 6163268 B/op	  300008 allocs/op
BenchmarkSringBuilder-12    	   10000	   3042595 ns/op	11671507 B/op	      35 allocs/op 
```

```
	for n := 0; n < b.N; n++ {
		for _, name := range inputList {
			name = fmt.Sprintf("%s%s", prefix, name)
		}
	}
```

```
        for n := 0; n < b.N; n++ {
		var sb strings.Builder
		for _, name := range inputList {
			sb.WriteString(prefix)
			sb.WriteString(name)
			name = sb.String()
		}
	}
```

in above len(inputlist) is 100000 with strings such as 		"gateway.tracing.", "gateway.commit", "gateway.processed", "gateway.sent", ....

Reply:
thanks for the report. i wasn't aware of the size and hence didn't want to add buffering overhead. can change it.

Reply:
the size/len of datapoints can be anywhere from 1 - 1M so having this can help on payloads which are closer to upper limit.

Reply:
yes you are right let me change the code a bit and re-run the benchmark.. Nice catch btw

Reply:
updated numbers
```
BenchmarkSprintf-12         	    3000	  16898194 ns/op	 6315346 B/op	  300008 allocs/op
BenchmarkSringBuilder-12    	   10000	   5848803 ns/op	 3912663 B/op	  200000 allocs/op
```

```
	for n := 0; n < b.N; n++ {
		for _, name := range inputList {
			var sb strings.Builder
			sb.WriteString(prefix)
			sb.WriteString(name)
			name = sb.String()
		}
	}
```
if you create sb for all iteration no need to reset